include/exadg/operators/variable_coefficients.h

```diff
@@ -235,6 +237,147 @@ class VariableCoefficients
   bool store_cell_based_face_data{false};
 };
 
+template<typename kernel_type>
```

Comment:
I would like to avoid this dependency of the analytical coefficient model on the kernel. The connection is actually the `VariableCoefficients` class and not the kernel in my opinion, just as you placed this functionality in the file variable_coefficients.h. I would not template `VariableCoefficients` here, but let the analytical model explicitly use the class `VariableCoefficients`.